lib/actions/tag.actions.ts

```diff
@@ -0,0 +1,77 @@
+import dbConnect from "@/database/dbConnect";
+import {
+  GetTagByPostIdParams,
+  GetTagByUserIdParams,
+  UpdateTagParams,
+  createNewTagParams,
+} from "@/database/models/shared.types";
+import tagModel, { ITag } from "@/database/models/tag.model";
+import { revalidatePath } from "next/cache";
+
+export const getTagByPostAndUserId = async ({
+  userId,
+  postId,
+}: {
+  userId: string;
+  postId: string;
+}) => {
+  try {
+    await dbConnect();
+    const tag = await tagModel.find({ user: userId, post: postId });
+    return tag as ITag[];
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const createNewTag = async (data: createNewTagParams) => {
+  try {
+    await dbConnect();
+    const tag = await tagModel.create({
+      name: data.name,
+      posts: data.postId,
+      users: data.userId,
+    });
+
+    return tag;
+  } catch (error) {
+    console.log(error);
+  }
+};
```

Comment:
We should test this. Should posts and users be arrays of id or just the IDs?

Reply:
we can talk during the meeting cause this actually was a harder decision to think about how it would work and I wasn't quite sure, but yes we should test it.